scripts/build_images.sh

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+cd "$(dirname "$0")"
+source "./lib.sh"
+
+check_dependencies \
+  docker
+
+source "./images.sh"
+
+PROJECT_ROOT="$(git rev-parse --show-toplevel)"
+TAG="ubuntu"
+DRY_RUN=false
+QUIET=false
+
+function usage() {
+  echo "Usage: $(basename "$0") [options]"
+  echo
+  echo "This script builds Coder's container images."
+  echo
+  echo "Options:"
+  echo " -h, --help                   Show this help text and exit"
+  echo " --dry-run                    Show commands that would run, but"
+  echo "                              do not run them"
+  echo " --tag=<tag>                  Select an image tag group to build,"
+  echo "                              one of: arch, centos, ubuntu)"
+  echo " --quiet                      Suppress container build output"
+  exit 1
+}
+
+# Allow a failing exit status, as user input can cause this
+set +o errexit
+options=$(getopt \
+            --name="$(basename "$0")" \
+            --longoptions=" \
+                help, \
+                dry-run, \
+                tag:, \
+                quiet" \
+            --options="h" \
+            -- "$@")
+# allow checking the exit code separately here, because we need both
+# the response data and the exit code
+# shellcheck disable=SC2181
+if [ $? -ne 0 ]; then
+  usage
+fi
+set -o errexit
+
+eval set -- "$options"
+while true; do
+  case "${1:-}" in
+  --dry-run)
+    DRY_RUN=true
+    ;;
+  --tag)
+    shift
+    TAG="$1"
+    ;;
+  --quiet)
+    QUIET=true
+    ;;
+  -h|--help)
+    usage
+    ;;
+  --)
+    shift
+    break
+    ;;
+  *)
+    # Default case, print an error and quit. This code shouldn't be
+    # reachable, because getopt should return an error exit code.
+    echo "Unknown option: $1"
+    usage
+    ;;
+  esac
+  shift
+done
+
+docker_flags=()
+
+if [ $QUIET = true ]; then
+  docker_flags+=(
+    --quiet
+  )
+fi
+
+for image in "${IMAGES[@]}"; do
+  image_dir="$PROJECT_ROOT/images/$image"
+  image_file="Dockerfile.$TAG"
+  image_ref="codercom/enterprise-$image:$TAG"
+  image_path="$image_dir/$image_file"
+
+  if [ ! -f "$image_path" ]; then
+    # echo "Path '$image_path' does not exist; skipping" >&2
```

Comment:
Left in but commented out because it's useful sometimes, but noisy... I don't mind removing either though

Reply:
Could we do a `QUIET` check for this? Also fine with how it is now.

Reply:
yea that's an option and what I had before, but then it'd show up in the build, and mostly it's ok. I don't have strong feelings about it either and would be happy to add it

like on a scale of 0 (leave it how it is) to 10 (def add a quiet check and add the log), where do you fall? I'm personally like right in the middle, which is useless for decision making :joy:

Reply:
Depends on the time. I think I'm at a 5.5 right now.

Reply:
nbd, will add it :smile: